### ISSUE TITLE
codewars kata: Simple Encryption #1 - Alternating Split

### ISSUE THREAD

**Issue by sulaimancode** [makersacademy/problem-solving#34]
Hey all, hope you're all doing well.

I've been working on the kata "Simple Encryption #1 - Alternating Split" on codewars, here is the link: https://www.codewars.com/kata/simple-encryption-number-1-alternating-split/train/ruby

I am stuck on how to decrypt a string when passed to the decrypt method. I pass all the tests for the encrypt method. Is there a good way to work backwards from a method in ruby? or do I need to figure out a way to reverse everything in the encrypt method?. I tried to reverse the encrypt method but still couldn't pass the tests.

Here is what I have for my ecrypt method:

```
def encrypt(text, n)
  return text if n <= 0
  return nil if text == nil
  return "" if text == ""
  arr = text.chars
  n.times do
    other_letter = arr.select.with_index { |e, i| i.odd? }
    remainder = arr.select.with_index { |e, i| i.even? }
    @result = other_letter.join << remainder.join
    arr = other_letter + remainder
  end
 @result
end

**Comment by ns-winter** [comment]
Hey Sol,

Good job with the encrypt part of the exercise.  I had a look at the challenge and tried to reverse the effects of the encrypt method and it seems to have worked; my code can be a lot cleaner but I'll put down what I came up with.

What helped me solve it was working it out on paper first to try and figure out what needed to be done and what steps needed to be taken. After that it wasn't too bad.  I'm pasting in your encrypt method along with my decrypt method below.  Please let me know if you have any queries re: the code and I'll do my best to try and explain what I did 😊👍

```
def encrypt(text, n) 
 return text if n <= 0 
  return nil if text == nil 
  return "" if text == "" 
  arr = text.chars 
  n.times do 
  other_letter = arr.select.with_index { |e, i| i.odd? } 
  remainder = arr.select.with_index { |e, i| i.even? } 
  @result = other_letter.join << remainder.join 
  arr = other_letter + remainder 
 end 
 @result 
end

def decrypt(encrypted_text , n)
  return encrypted_text if n <=0
  return nil if encrypted_text == nil
  return "" if encrypted_text == ""
  x = encrypted_text
  n.times do
    answer = []
    part1, part2 = x.slice!(0...(x.length/2)).split(''), x.split('')
      while part2.count >= 1 do
        answer = answer << part2.shift << part1.shift
        x = answer.join
      end
  end
  x
end
```

**Comment by sulaimancode** [comment]
That's a very clever way of doing it, I can see what you've done for your decrypt method. Thank you for sharing it with me, I tried different ways of reversing it but couldn't figure out the right way.

Did you have a different encrypt method?

**Comment by ns-winter** [comment]
Hey, thanks and no worries.  No I used your encrypt method - only saw the exercise from your post but it looked interesting 😃- so I decided the work on the decrypt method.

As I said, I worked it out on paper using the three answer/examples in the description and tracked the change in movement of the individual elements using their original index locations. That helped determine the pattern that the decrypt code needed to follow.
